client/index.js

```diff
@@ -59,10 +59,18 @@ var onSocketMsg = {
 };
 
 var newConnection = function() {
+	var hostname = urlParts.hostname;
+
+	if (urlParts.hostname === '0.0.0.0') {
+		if (window.location.hostname && !!~window.location.protocol.indexOf('http')) {
```

Comment:
Why the check for `http` here (not saying it is wrong per se)? If it has a valid reason, could you add a comment above? And also please give an example for when the hostname does not exist in the comments.

We get so many PR's around the detection of this url, so comments will help other people here.

Reply:
I have a problem with this part as I need host '0.0.0.0' but socket port from urlParts.port, not from self.location.port. Am I right that I can't setup socket port explicitly?